gpii/node_modules/solutionsRegistry/src/js/filtered-settings-middleware.js

```diff
@@ -0,0 +1,174 @@
+        "rules.filterInput": "nomerge"
+    },
+    settingsSchemaPath: "%gpii-universal/build/schemas/settings-schema.json",
+    settingsSchema: "@expand:gpii.universal.solutionsRegistry.getSettingsSchema({that}.options.settingsSchemaPath)",
+    filteredSettingsSchema: "@expand:gpii.universal.solutionsRegistry.kettle.request.http.generateFilteredSchema({that}.options.settingsSchema, {request}.req, {that}.options.rules.filterInput)",
+    rules: {
+        filterInput: "{that}.options.rules.requestContentToValidate"
+    },
+    components: {
+        validationMiddleware: {
+            options: {
+                inputSchema: "{gpii.universal.solutionsRegistry.kettle.request.http.base}.options.inputSchema"
+            }
+        }
+    }
+});
+
+fluid.defaults("gpii.universal.solutionsRegistry.kettle.request.http.settings", {
+    gradeNames: ["gpii.universal.solutionsRegistry.kettle.request.http.base"],
+    rules: {
+        filterInput: {
+            "": "body.flat.contexts.gpii-default.preferences"
+        }
+    },
+    inputSchema: {
+        type: "object",
+        properties: {
+            flat: {
+                type: "object",
+                required: true,
+                properties: {
+                    contexts: {
+                        type: "object",
+                        required: true,
+                        additionalProperties: {
+                            type: "object",
+                            required: true,
+                            properties: {
+                                settings: "{that}.options.filteredSettingsSchema"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+});
+
+fluid.defaults("gpii.universal.solutionsRegistry.kettle.request.http.preferences", {
+    gradeNames: ["gpii.universal.solutionsRegistry.kettle.request.http.base"],
+    rules: {
+        filterInput: {
+            "": {
+                transform: {
+                    type: "fluid.transforms.firstValue",
+                    // TODO: discuss how to handle contexts other than gpii-default.
+                    values: [
+                        "body.contexts.gpii-default.preferences.application",
+                        "body.contexts.gpii-default.preferences"
+                    ]
+                }
+            }
+        }
+    },
+    inputSchema: {
+        type: "object",
+        properties: {
+            body: {
+                type: "object",
+                properties: {
+                    contexts: {
+                        type: "object",
+                        required: true,
+                        minProperties: 1,
+                        additionalProperties: {
+                            type: "object",
+                            properties: {
+                                metadata: {
+                                    type: "array",
+                                    items: {
```

Comment:
This is probably the appropriate point to factorise this schema into something which specifies a "metadataItem" or so

Reply:
I left a TODO there, as we'd need one or more of the proposed inheritance strategies to do this cleanly.